File: binarySearch/binarySearch.js
function binarySearch(arr, target) {
    let first = 0;
    let last = arr.length - 1;
    let position = -1;
    let found = false;
    let middle;

    while (found === false && first <= last) {
        middle = Math.floor((first + last) / 2);
        if (arr[middle] == target) {
            found = true;
            position = middle;
        }else if (arr[middle] > target){
            last = middle - 1;
        }else{
            first = middle + 1;
        }
    }
    return position;
}

binarySearch([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20], 13);

File: binarySearch/binarySearch.py
def binarySearch(alist, item):
    first = 0
    last = len(alist)-1
    midpoint = 0

    while first <= last:
        midpoint = (first + last)//2
        
        if alist[midpoint] < item:
            first = midpoint + 1
        elif alist[midpoint] > item:
            last = midpoint - 1
        else:
            return midpoint
    return -1


test = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
result = binarySearch(test, 11)


if result != -1:
    print("Element is present at ", str(result))
else:
    print('Element is not present')
